src/app/services/api.service.ts

```diff
@@ -1,105 +1,98 @@
+import { Observable, throwError } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
 import { Injectable } from '@angular/core';
-import { Http, Headers, Response, RequestOptions } from '@angular/http';
-import { Router } from '@angular/router';
-import { Observable } from 'rxjs/Observable';
+import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
 import { environment } from '../../environments/environment';
 import 'rxjs/Rx';
 
 @Injectable()
 export class APIService {
 
-    private _headers: Headers;
-    private _localStorageKey = environment.localStorageKey;
 
-    constructor(private _http: Http) {}
+  private _localStorageKey = environment.localStorageKey;
 
-    /**
-     * Set generic headers for all requests
-     */
-    private _setHeaders() {
-        this._headers = new Headers();
-        this._headers.append('Content-Type', 'application/json');
-        this._headers.append('Accept', 'application/json');
-        this.setTokenHeader();
-    }
 
-    /**
-     * Perform GET request
-     * @param  {string}          url URL of the API
-     * @return {Observable<any>}     Return response
-     */
-    public get(url: string): Observable<any> {
-        this._setHeaders();
-        return this._http.get(url, { headers: this._headers })
-                    .catch(this._handleError);
-    }
+  constructor(private _http: HttpClient) {
+  }
 
-    /**
-     * Perform POST request
-     * @param  {string}          url  URL of the API
-     * @param  {Object}          body Request body
-     * @return {Observable<any>}      Return response
-     */
-    public post(url: string, body: Object): Observable<any> {
-        this._setHeaders();
-        return this._http.post(url, body, { headers: this._headers })
-                    .catch(this._handleError);
-    }
+  /**
+   * Perform GET request
+   * @param  {string}          url URL of the API
+   * @return {Observable<any>}     Return response
+   */
+  public get(url: string): Observable<any> {
+    return this._http.get(url, {headers: {'Content-Type': 'application/json', 'Accept' : 'application/json', 'x-access-token' : localStorage.getItem(this._localStorageKey) + ''}}).pipe(
```

Comment:
Is the old way not working.. this seems to be very dirty.. If the previous issue was not because of this then revert it.